packages/mds-web-sockets/client.ts

```diff
@@ -13,29 +14,43 @@ async function sendAuth() {
   return connection.send(`AUTH%Bearer ${TOKEN}`)
 }
 
-function getClient() {
+async function getClient() {
   if (connection && connection.readyState === 1) {
     return connection
   }
-  connection = new WebSocket(URL)
 
-  setWsHeartbeat(connection as WebSocketBase, 'PING')
+  try {
+    const res = await requestPromise(`http://${URL}`)
 
-  connection.onopen = async () => {
-    await sendAuth()
-  }
+    if (res.statusCode === 503) {
+      throw new Error('Could not connect to WebSocket server')
+    }
+    connection = new WebSocket(`ws://${URL}`)
 
-  connection.onerror = async err => {
-    return log.error(err)
-  }
+    setWsHeartbeat(connection as WebSocketBase, 'PING')
+
+    connection.onopen = async () => {
+      await sendAuth()
+    }
+
+    connection.onerror = async err => {
```

Comment:
I know this was already here, but I don't think this function needs an `async` modifier